src/components/CommentForm.vue

```
@@ -0,0 +1,140 @@
+        ></textarea>
+      </div>
+      <p v-if="submitted && !body" class="help is-danger">Comment text is required</p>
+    </div>
+
+    <div class="field is-grouped">
+      <div class="control">
+        <button
+          type="submit"
+          class="button is-link"
+          :class="{ 'is-loading': loading }"
+        >
+          Submit
+        </button>
+      </div>
+      <div class="control">
+        <button
+          type="button"
+          class="button is-light"
+          @click="clearForm"
+          :disabled="loading"
+        >
+          Clear
+        </button>
+      </div>
+    </div>
+  </form>
+</template>
+
+<script>
+export default {
+  name: 'CommentForm',
+  props: {
+    postId: {
+      type: Number,
+      required: true,
+    },
+    onSubmit: {
+      type: Function,
+      required: true,
+    },
+  },
+  data() {
+    return {
+      name: '',
+      email: '',
+      body: '',
+      loading: false,
+      submitted: false,
+    };
+  },
+  methods: {
+    clearForm() {
+      this.name = '';
+      this.email = '';
+      this.body = '';
+      this.submitted = false;
+    },
+
+    async handleSubmit() {
+      this.submitted = true;
+
+      if (!this.name || !this.email || !this.body) {
+        return;
+      }
+
+      this.loading = true;
+
+      try {
+        const newComment = await this.onSubmit({
+          name: this.name,
+          email: this.email,
+          body: this.body,
+        });
+
+        // mantém nome e e-mail, limpa apenas o texto
+        this.body = '';
+        this.submitted = false;
+
+        // opcional: foco no campo de comentário novamente
+        this.$nextTick(() => {
+          const textarea = this.$el.querySelector('textarea');
+          if (textarea) textarea.focus();
+        });
+
+        // append é feito no App.vue ou componente pai
+      } catch (error) {
+        console.error('Failed to submit comment', error);
+      } finally {
+        this.loading = false;
```

Current catch block swallows the error (logs it) but does not propagate it to the parent. The optional requirement asks that Add errors be handled so the user can retry. Emit an `error` event or rethrow the error after logging so the parent can show a retry UI / notification. For example: `this.$emit('error', error)` (or `throw error`) so the parent can react. Without this, users cannot retry failed adds.